tooling/templatize/pkg/pipeline/run.go

```diff
@@ -0,0 +1,190 @@
+package pipeline
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"text/template"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/containerservice/armcontainerservice"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/resources/armsubscriptions"
+	"gopkg.in/yaml.v3"
+
+	"github.com/Azure/ARO-HCP/tooling/templatize/pkg/config"
+)
+
+func LoadPipeline(path string, vars config.Variables) (*Pipeline, error) {
+	tmpl := template.New("pipeline")
+	content, err := os.ReadFile(path)
+	if err != nil {
+		return nil, err
+	}
+
+	tmpl, err = tmpl.Parse(string(content))
+	if err != nil {
+		return nil, err
+	}
+
+	var tmplBytes bytes.Buffer
+	if err := tmpl.Option("missingkey=error").Execute(&tmplBytes, vars); err != nil {
+		return nil, err
+	}
+
+	pipeline := &Pipeline{}
+	err = yaml.Unmarshal(tmplBytes.Bytes(), pipeline)
+	if err != nil {
+		return nil, err
+	}
+	return pipeline, nil
+}
+
+func (p *Pipeline) Run(vars config.Variables) error {
+	for _, step := range p.Steps {
+		err := step.run(vars)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (s *Step) run(vars config.Variables) error {
+	switch s.Action.Type {
+	case "Shell":
+		return s.runShell(vars)
+	default:
+		return fmt.Errorf("unsupported action type %q", s.Action.Type)
+	}
+}
+
+func (s *Step) runShell(vars config.Variables) error {
+	fmt.Printf("Running step %s: %q\n", s.Name, s.Action.Command)
+	fmt.Printf("Resource Group %s\n", s.ResourceGroup)
+	fmt.Printf("Subscription %s\n", s.Subscription)
+
+	// lookup subscription ID via name using the azure sdk
+	subscriptionID, err := lookupSubscriptionID(s.Subscription)
+	if err != nil {
+		return fmt.Errorf("failed to lookup subscription ID: %w", err)
+	}
+
+	// prepare kubeconfig
+	kubeconfigFile, err := buildAKSKubeConfig(
+		subscriptionID,
+		s.ResourceGroup,
+		"aro-hcp-aks",
+		context.Background(),
+	)
+	if err != nil {
+		return fmt.Errorf("failed to build kubeconfig: %w", err)
+	}
+
+	// schedule the deletion of the kubeconfig file after the command execution
+	defer func() {
+		if err := os.Remove(kubeconfigFile); err != nil {
+			fmt.Printf("Warning: failed to delete kubeconfig file %s: %v\n", kubeconfigFile, err)
+		}
+	}()
+
+	// build ENV vars
+	envVars := os.Environ()
+	envVars = append(envVars, fmt.Sprintf("KUBECONFIG=%s", kubeconfigFile))
+	for _, e := range s.Action.Variables {
+		value := vars[e.Value] // todo nested lookups
+		envVars = append(envVars, fmt.Sprintf("%s=%s", e.Name, value))
+	}
+
+	// execute the shell command with the environment variables
+	cmd := exec.Command("/bin/sh", "-c", s.Action.Command)
```

Comment:
I don't think it benefits us to assume this - the Ev2 configuration can support whatever. We should just run `[]string` into `exec` and call it a day.

Reply:
done